Framework/CorDebug/VsProject.cs

```diff
@@ -142,11 +143,24 @@ public string[] GetDependencies( bool fIncludeStartProgram, bool fPE, bool fTarg
                     AddDependencies( table, new string[] { this.m_startProgam } );
                     AddDependencies( table, this.m_startProgramReferences );
                 }
+                // if mscorlib isn't in the list of assemblies, add it as it's always required
+                // the build system in some versions of VS+MSBuild will assume that, while other
+                // versions will explicitly reference it with the InProject item property set to
+                // false so it won't show in the IDE. By testing for it and adding it if not found
+                // both cases are covered. 
+                var q = from path in table.OfType<string>( )
+                        where 0 == string.Compare( Path.GetFileName( path ), "mscorlib.dll", StringComparison.InvariantCultureIgnoreCase )

```

Comment:
I believe this should be [OrdinalIgnoreCase](https://blogs.msdn.microsoft.com/bclteam/2007/05/31/string-compare-string-equals-josh-free/) comparison, invariant culture considers things equal that file system does not. And you can use Equals(string, string, StringComparison) for better readability.